app/services/hyrax/characterization/valkyrie_characterization_service.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+require 'hydra-file_characterization'
+require 'nokogiri'
+
+class Hyrax::Characterization::ValkyrieCharacterizationService
+  ##
+  # @api public
+  # @param [Hyrax::FileMetadata] metadata  which has properties to recieve characterization values
+  # @param [Valkyrie::StorageAdapter::StreamFile] source to run characterization against
+  # @param [Hash] options the options pass to characterization
+  # @option options [Hash{Symbol => Symbol}] parser_mapping
+  # @option options [Class] parser_class
+  # @option options [Symbol] ch12n_tool
+  #
+  # @return [Hash]
+  def self.run(metadata, source = nil, **options)
```

Comment:
i took the liberty of removing positional arguments from the initializer, but wasn't sure if `#run` has an interface defined upstream somewhere.

does this class have potential callers outside the current diff, or should we use named parameters here too?